policy/release/olm/olm.rego

```diff
@@ -175,6 +175,44 @@ deny contains result if {
 	result := lib.result_helper_with_term(rego.metadata.chain(), [component.containerImage], component.containerImage)
 }
 
+# METADATA
+# title: Unpinned related images for a component
+# description: >-
+#   Check the input image for the presence of related images.
+#   Ensure the images for the presence of unpinned image references
+# custom:
+#   short_name: unpinned_related_images
+#   failure_msg: The %q related image reference is not pinned.
+#   solution: >-
+#     Update the related images replacing the unpinned image reference
+#     with pinned image reference. Pinned image reference contains the image digest
+#   collections:
+#   - redhat
+#
+deny contains result if {
+	_release_restrictions_apply
+
+	snapshot_components := input.snapshot.components
+	component_images_digests := [component_image.digest |
+		some component in snapshot_components
+		component_image := image.parse(component.containerImage)
+	]
+
+	some related_images in _related_images(input.image)
+
+	unmatched_image_refs := [related |
+		some related in related_images
+		not related.digest in component_images_digests
+	]
+
+	some unmatched_image in unmatched_image_refs
+	unmatched_image.digest == "" # unpinned image references have no digest
+	unmatched_ref := sprintf("%s:%s", [unmatched_image.repo, unmatched_image.tag])
+	not ec.oci.descriptor(unmatched_ref)
```

Comment:
I think this line should be removed. IIUC, this is included in the other check as the way to determine if the image is accessible. So this is an expensive query to the registry, and we don't need it here when we just want to see if the image ref includes a digest.